src/analyserequation.cpp

```diff
@@ -68,6 +68,29 @@ AnalyserEquation::Type AnalyserEquation::type() const
     return mPimpl->mType;
 }
 
+std::string AnalyserEquation::typeAsString() const
+{
+    if (mPimpl->mType == Type::TRUE_CONSTANT) {
+        return "TRUE_CONSTANT";
```

Comment:
Are we looking for a direct string form here or a more readable form "True constant"?

Reply:
This is open for debate. Truth be told, when I first started working on this PR, I had a "proper" string for each `enum class` then when I got to `AnalyserEquationAst`'s `enum class`, I decided to simply have a direct string form. I mean, for `AnalyserEquationAst::Type::ACSCH`, do we really want something like `The arc hyperbolic cosecant function.`? Also, should we start with a capital letter? Should have the `the` article? Should we end up with a full stop? If anything, I might want `arc hyperbolic cosecant function`, but that's rather long.

Otherwise, I have just noticed that for `version()`, we have `versionString()` as opposed to `versionAsString()`. So, if anything, we might to rename the current `xxxAsString()` to `xxxString()`?

Also, looking at https://github.com/cellml/libcellml/pull/1132, I can't say that I am keen on `enumerationAsString()`. It's not part of a class itself, which means that an IDE won't suggest when we type something like `myAnalyserModel->type...` and, instead, the user needs to know about it (who reads documentation...). (Also, had I know that such a PR was in the work (no associated issue?), I wouldn't have wasted my time on this PR...)

Reply:
It wasn't in the works, I just wanted to try out overloading on enumerations. with `enumerationAsString` once I know it is there I can use it for any and all enumerations.

I don't think it should be part of the class, it doesn't sit right for me.

I would probably argue that we should change to `xxxAsString()` rather than the alternative.

Reply:
I wouldn't go for super long names myself the shorter forms are acceptably common, perhaps more so than the formal term.

Reply:
There doesn't appear to be an associated issue with this PR?